docs/http_api.rst

```diff
@@ -0,0 +1,1035 @@
+   Link: <https://example.com/endpoint/?page=2>; rel="next"
+
+   [....]
+
+.. _tokens:
+
+Tokens
+^^^^^^
+
+For the token endpoints, no authentication is required.
+
+.. http:post:: /tokens/
+
+   Create a new token for the provided user.
+
+   :<json str username: The username of the user to create the token for.
+   :<json str password: The password of the user to create the token for.
+   :>json str token: The generated token.
+   :status 201: When the token is successfully generated.
+   :status 400: When the user credentials are incorrect.
+
+   **Example request**:
+
+   .. sourcecode:: http
+
+      POST /tokens/ HTTP/1.1
+      Content-Type: application/json
+
+      {"username":"testuser","password":"testpassword"}
+
+   
+   **Example response**:
+
+   .. sourcecode:: http
+
+      HTTP/1.1 201 Created
+      Content-Type: application/json
+
+      {"token":"eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWV9.TJVA95OrM7E2cBab30RMHrHDcEfxjoYZgeFONFh7HgQ"}
+
+
+.. http:get:: /token
+
+   Verify that an existing token is valid, and return the token's payload.
+
+   :>header Authorization: "JWT " followed by the token to verify
+   :>json obj payload: The payload of the token.
+   :status 200: The token is valid.
+   :status 400: The token is invalid.
+
+   **Example request**:
+
+   .. sourcecode:: http
+   
+      GET /token HTTP/1.1
+      Authorization: JWT eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiIxMjM0NTY3ODkwIiwibmFtZSI6IkpvaG4gRG9lIiwiYWRtaW4iOnRydWV9.TJVA95OrM7E2cBab30RMHrHDcEfxjoYZgeFONFh7HgQ
+
+
+   **Example response**:
+
+   .. sourcecode:: http
+
+      HTTP/1.1 200 OK
+      Content-Type: application/json
+
+      {"payload":{"sub":"1234567890","name":"John Doe","admin":true}}
+
+
+Password resets
+^^^^^^^^^^^^^^^
+
+For the password reset endpoints, no authentication is required.
+
+To reset a user's password, the following steps should be followed:
+
+1. Make a request to the reset endpoint.
+   This will make an HTTP request to the preconfigured endpoint with the user's
+   details, and a token.
+2. Make a request to the confirm endpoint, with the provided token and the new
+   password.
+
+.. http:post:: /passwords/resets/
+
+   Start the process for resetting a user's password.
+
+   :<json str username: The username of the user to reset the password for.
+   :<json str app:
+        The application that the token should go to, configured in settings.
+   :code 202: The password reset process was started.
+   :code 400: The username does not exist.
+
+   **Example request**:
+
+   .. sourcecode:: http
+
+      POST /passwords/resets/ HTTP/1.1
+      Content-Type: application/json
+
+      {"username":"jonsnow","app":"numi"}
+
+   **Example response**:
+
+   .. sourcecode:: http
+      
+      HTTP/1.1 202 Accepted
+
+.. http:post:: /passwords/confirmations/
+
+   Reset the users password using the provided token.
+
+   :<json str token: The provided token.
```

Comment:
Vaguely remember us discussing a reason for putting the token in the body here, but wondering if it would be more consistent to include the token in the header.